### PR TITLE
Fix relative paths

### PR DESCRIPTION
**Pull request by fjtrujy**
## Description
This PR improves the method that transforms the relative path into absolute paths, something needs to support POSIX functions with relative paths.

Specific content for `cwd` and paths has been moved to a separate file.

The returned error for several functions has been improved.

Once again the base code has been taken from the PSPSDK.

Cheers.

### CODE REVIEW

File: ee/libcglue/src/glue.c [hunk 1]
```diff
@@ -134,132 +130,17 @@ void compile_time_check() {
...
-				strcpy(b_fname, "host:");
-				strcpy(b_fname + 5, buf);
-			}
-		} else {                 // otherwise, it's relative directory, let's copy pwd straight
-			int b_fname_len = strlen(__direct_pwd);
-			if (!strchr(__direct_pwd, ':')) { // check if pwd contains device name
-				strcpy(b_fname, "host:");
-				strcpy(b_fname + 5, __direct_pwd);
-				if (!(__direct_pwd[b_fname_len - 1] == '/' || __direct_pwd[b_fname_len - 1] == '\\')) { // does it has trailing slash ?
-					if(isCdromPath(b_fname)) {
```
Comment (fjtrujy) [inline ee/libcglue/src/glue.c:224]: Is this thing really needed?
I don't know why but it compares If unit is `cdrom` and then adding `\\` is that something legacy? maybe from windows?

@uyjulian @rickgaiser , do you know something about this?

File: ee/libcglue/src/glue.c [hunk 2]
```diff
@@ -276,7 +157,7 @@ int _open(const char *buf, int flags, ...) {
 		is_dir = 1;
 	}
 
-	char *t_fname = __normalized_path(buf);
+	printf("open: %s, buf: %s\n", t_fname, buf);
```
[the remark this answers is not in the document]
Comment (fjtrujy) [reply ee/libcglue/src/glue.c]: Thanks! Removed!

File: ee/libcglue/src/glue.c (hunk 1 above)
Comment (uyjulian) [reply ee/libcglue/src/glue.c:224]: This is the correct behavior. In the versions of cdvdman I've checked `CdSearchFile` checks for backslashes (and not forward slashes)

**Review by uyjulian — APPROVED** [review]
Lgtm

**Pull request merged**